velox/type/Timestamp.h

```diff
@@ -31,26 +31,34 @@ class time_zone;
 namespace facebook::velox {
 
 struct TimestampToStringOptions {
-  enum Precision : int8_t {
+  enum class Precision : int8_t {
     kMilliseconds = 3,
     kNanoseconds = 9,
-  } precision = kNanoseconds;
+  };
+
+  Precision precision = Precision::kNanoseconds;
 
   bool zeroPaddingYear = false;
   char dateTimeSeparator = 'T';
-  bool dateOnly = false;
-};
 
-// Our own version of gmtime_r to avoid expensive calls to __tz_convert.  This
-// might not be very significant in micro benchmark, but is causing significant
-// context switching cost in real world queries with higher concurrency (71% of
-// time is on __tz_convert for some queries).
-//
-// Return whether the epoch second can be converted to a valid std::tm.
-bool epochToUtc(int64_t seconds, std::tm& out);
+  // When TimestampToStringOptions.mode = Mode::kFull, return ISO 8601
+  // timestamp format, %Y-%m-%dT%H:%M:%S.nnnnnnnnn for nanoseconds
+  // precision, or %Y-%m-%dT%H:%M:%S.nnn for milliseconds precision.
+  //
+  // When TimestampToStringOptions.mode = Mode::kDateOnly, return ISO 8601
+  // date format(%Y-%m-%d).
+  //
+  // When TimestampToStringOptions.mode = Mode::kTimeOnly, return ISO 8601
+  // time format, %H:%M:%S.nnnnnnnnn for nanoseconds precision,
+  // or %H:%M:%S.nnn for milliseconds precision.
+  enum class Mode : int8_t {
+    kFull,
```

Comment:
Consider moving comments next to enum values:

```
/// ISO 8601 timestamp format: %Y-%m-%dT%H:%M:%S.nnnnnnnnn for nanoseconds precision; %Y-%m-%dT%H:%M:%S.nnn for milliseconds precision.
kFull,

/// ISO 8601 date format: %Y-%m-%d.
kDateOnly,

/// ISO 8601 time format:...
kTimeOnly,
```

Reply:
Fixed